composer.json

```diff
@@ -14,8 +14,11 @@
   "require-dev": {
     "dealerdirect/phpcodesniffer-composer-installer": "^0.7",
     "phpcompatibility/php-compatibility": "^9.3",
+    "phpstan/extension-installer": "^1.3.0",
+    "phpstan/phpstan": "^1.10",
     "phpunit/phpunit": "^4|^5|^6|^7|^8|^9",
     "squizlabs/php_codesniffer": "^3.5",
+    "szepeviktor/phpstan-wordpress": "^1.3.0",
```

Comment:
Not that this package and PHPStan itself require PHP 7.2.

So when we start running tests against PHP 5.6 (see #399), `composer install` would break.

The plugin itself wouldn't break, as it doesn't use the autoloader.

Same discussion as in #729 really.

Check out https://github.com/10up/plugin-check/pull/166 for a workaround by moving these deps to a separate `composer.json` file.

Reply:
Great point @swissspidy, let's apply the same approach here as we want to support PHP 5.6 too for our test suite as we should preferably run the tests on that version too.

Reply:
How about we address back-compat with PHP 5.6 as part of #399?

Reply:
Downsides of (2) and (3) are that it's a less reproducible setup for developers, and that it's cumbersome to do for other packages as well (think all the PHPStan extensions or the Slevomat CS package). I did something like that in the past for downgrading/upgrading PHPUnit in CI, but it's not more maintainable than (1).

I wish (1) wasn't needed, but when dealing with such version requirements it's a necessary evil.

> Dependabot wouldn't automatically be aware of it.

FWIW this is easily solved with another entry in the Dependabot config. Example:

```yml
  - package-ecosystem: "composer"
    directory: "/"
    schedule:
      interval: "daily"

  - package-ecosystem: "composer"
    directory: "/build-cs"
    schedule:
      interval: "daily"
```

Reply:
> 3\. In GHA workflow, just-in-time `composer remove` this package before doing `composer install` prior to running PHPUnit in PHP<7.2.

I just noticed that Gutenberg actually [takes this approach](https://github.com/WordPress/gutenberg/blob/064f7319b8f7a8860ef4d295e37a53e46bcebae6/.github/workflows/unit-test.yml#L118-L140) instead of maintaining a separate `composer.json`.

Reply:
How is this less maintainable than maintaining a separate `composer.json`?